docs/articles/commands/text/custom_prefix_handler.md

```diff
@@ -0,0 +1,99 @@
+    // and to the "!" prefix.
+    // If you want to change it, you first set if the bot should react to mentions
+    // and then you can provide as many prefixes as you want.
+    PrefixResolver = prefixResolver.ResolvePrefixAsync
+});
+```
+
+In this example, the bot will react to mentions and to the `!` and `?` prefixes. The `true` argument in the `DefaultPrefixResolver` constructor specifies that the bot should react to mentions.
+
+## IPrefixResolver
+
+The `IPrefixResolver` interface is a bit more complex to set up, but can make dynamic prefixes easier overall. The interface is as follows:
+
+```cs
+public interface IPrefixResolver
+{
+    ValueTask<int> ResolvePrefixAsync(CommandsExtension extension, DiscordMessage message);
+}
+```
+
+On the surface, this is very similar to using the delegate method, with one important change: you can implement this interface.
+One may prefer this approach as opposed to setting the `PrefixResolver` property because implementing the interface allows you to participate in dependency injection.
+
+A common scenario is using a database to retrieve a per-server prefix. 
+Here's an example of how you can implement the `IPrefixResolver` interface:
+
+```cs
+public class CustomPrefixResolver(IDatabaseService database) : IPrefixResolver
+{
+    public async ValueTask<int> ResolvePrefixAsync(CommandsExtension extension, DiscordMessage message)
+    {
+        if (string.IsNullOrWhiteSpace(message.Content))
+        {
+            return -1;
+        }
+        // Mention check
+        else if (this.AllowMention && message.Content.StartsWith(extension.Client.CurrentUser.Mention, StringComparison.OrdinalIgnoreCase))
+        {
+            return extension.Client.CurrentUser.Mention.Length;
+        }
+        
+        // Database check
+        string prefix = await database.GetPrefixAsync(message.Channel.GuildId);
+        
+        if (message.Content.StartsWith(prefix, StringComparison.OrdinalIgnoreCase))
+        {
+            return prefix.Length;
+        }
+        
+        return -1;
+    }
+}
+```
+
+Now, unlike the normal prefix resolver delegate, this isn't set on the TextCommandConfiguration. Instead, you'll register this class with your service provider. This allows you to inject dependencies into your prefix resolver.
+
+> [!IMPORTANT]
+> The prefix resolver is resolved from a scoped service provider. For most scenarios, the only stipulation is that state should be held in an external, more persistent (e.g. singleton) service. Users of EntityFramework Core users should ensure that the DbContext is scoped correctly.
+
+```cs
+DiscordClientBuilder builder = DiscordClientBuilder.CreateDefault(discordToken, TextCommandProcessor.RequiredIntents | SlashCommandProcessor.RequiredIntents)
+.ConfigureServices
+(
+    services =>
+    {
+        services.AddScoped<IPrefixResolver, CustomPrefixResolver>();
+    }
+);
+```
+
+And just like that, you're off to the races.
+
+---
```

Comment:
unnecessary separator

Reply:
Was done to be consistent with the migration article, oops